custom_components/google_home/config_flow.py

```diff
@@ -51,28 +51,58 @@ async def async_step_user(
             return self.async_abort(reason="single_instance_allowed")
 
         if user_input is not None:
-            username = user_input[CONF_USERNAME]
-            password = user_input[CONF_PASSWORD]
             session = async_create_clientsession(self.hass)
-
-            if len(password) < MAX_PASSWORD_LENGTH:
+            try:
+                username = user_input[CONF_USERNAME]
+                password = user_input[CONF_PASSWORD]
+            except KeyError:
+                # either no username or password provided
+                username = None
+                password = None
+            
+            try:
+                master_token = user_input[CONF_MASTER_TOKEN]
+            except KeyError:
+                # no master token provided
+                master_token = None
+
+            if master_token is None and username is None:
+                self._errors["base"] = "missing-inputs"
+                return await self._show_config_form()
```

Comment:
```suggestion
            if not master_token or (not username and not password):
                self._errors["base"] = "missing-inputs"
                return await self._show_config_form()
```

Reply:
Needed to change the check here because this one was not working (always failing if some inputs are left empty)